packages/nimble-components/src/combobox/specs/IxD.md

```diff
@@ -0,0 +1,241 @@
+`ENTER` on the focused input field commits the typed, autocompleted, or selected value and closes the dropdown (if open). The caret remains visible after commit.
+
+![ ](spec-images/combobox-20.png)
+
+`ESC` clears any text in the field input.
+
+![ ](spec-images/combobox-21.png)
+
+Standard single line text editing keys, for example `DELETE` and `SHIFT` + `RIGHT ARROW`, should work when focus is in the field input (not pictured).
+
+#### Dropdown
+
+| Key                           | Description                                                                                                                                        |
+| ----------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `ENTER`                       | Sets the value of the field input to the focused list item, closes the dropdown, and moves focus to the field input                                |
+| `DOWN ARROW` or `UP ARROW`    | Moves visual focus to the next or previous list item                                                                                               |
+| `RIGHT ARROW` or `LEFT ARROW` | Moves visual focus to the field input and inserts the caret one character to the right or left                                                     |
+| `ESC`                         | Closes the dropdown and moves visual focus to the field input                                                                                      |
+| `HOME` or `END`               | Moves visual focus to the field input and inserts the caret at the beginning or end of the string                                                  |
+| _Printable characters_        | Moves visual focus to the field input, types the character(s), opens the autocomplete dropdown and/or displays inline autocomplete (if applicable) |
+
+From a focused list item, `DOWN ARROW` or `UP ARROW` moves focus and selects the next or previous list item. If the focused/selected list item is the first or last in the dropdown, `DOWN ARROW` or `UP ARROW` wraps to last or first list item (not pictured).
+
+![ ](spec-images/combobox-22.png)
+
+`ENTER` on a focused/selected list item commits the value, updates the field input, and closes the dropdown.
+
+![ ](spec-images/combobox-23.png)
+
+`ESC` on a focused/selected list item cancels the value, closes the dropdown, and moves visual focus to the field input.
+
+![ ](spec-images/combobox-24.png)
+
+`RIGHT ARROW` or `LEFT ARROW` moves visual focus to the field input without closing the dropdown and moves the caret one character to the right or left.
+
+![ ](spec-images/combobox-25.png)
+
+## Future Considerations
+
+-   Helper text for input components (where it should be located, how it should be accessed).
+-   Do clients want an "Add new" list item option? Do we want to document how clients might show an "Add new" list item indicator in the dropdown?
```

Comment:
I feel like this is more relevant to the filterable select. The combobox is a text field, you can type whatever value you want. At the very least it applies to both combobx and filterable select.
It does open some questions like would that button / option exist under each group (i.e. how do you choose what group to add the option to)? Or would there just be a generic add option command at the end of the list and the client should show some other UI (dialog, slideout, etc) for choosing the group, etc.

Reply:
I'd argue that this _can't_ be used for a filterable select. Ultimately in a select (at least with the typical usage of it in something like Angular) there is a model value associated with each option. There would be no good way to provide this feature that would allow a user to associate an appropriate model with a new option.

Reply:
Implementation-wise I think clicking that button just fires an event, I wouldn't expect it to mutate any configured state. It's just a purpose specific action button.

Reply:
How would an end-user add a new list option if they can't type the value into a select?

Reply:
It would have to be an auxiliary UI away from the select component itself. This is another reason that I'm not entirely keen on this as a feature. The user would not only have to supply the text to display in the dropdown, but also the model to use for that option.